specifyweb/frontend/js_src/lib/components/Merging/InvalidMergeRecords.tsx

```diff
@@ -16,24 +16,24 @@ import { TableIcon } from '../Molecules/TableIcon';
 
 function InvalidMergeRecords({
   resources,
-  header = '',
+  tableName,
+  body = '',
 }: {
+  readonly tableName: keyof Tables;
   readonly resources: RA<SerializedResource<AnySchema>>;
-  readonly header?: string;
+  readonly body?: string;
 }): JSX.Element {
   return (
     <div className="flex flex-1 flex-col gap-4">
-      <H2>{header}</H2>
+      <H2>{body}</H2>
       <Ul className="gap-2">
         {resources.map((resource, index) => (
           <li
             key={(resource.id as number) ?? index}
-            className="flex flex min-h-[theme(spacing.8)] flex-1 items-center gap-2"
+            className="flex min-h-[theme(spacing.8)] flex-1 items-center gap-2"
           >
-            <TableIcon label={true} name={resource._tableName} />
-            <FormattedResource
-              resource={deserializeResource(resource)}
-            ></FormattedResource>
+            <TableIcon label={true} name={tableName} />
+            <FormattedResource resource={deserializeResource(resource)} />
```

Comment:
(optional) use a useMemo. deserializeResource is somewhat expensive (serialize resource too, because it tries to convert from lower case back to camel case, which requires looping over every field in the table, for each field in the resource, thus `O(n^2)` at very least, but then you do this for every resource since you are in a loop, and you get `O(n^3)` 🤯)

Reply:
Resolved!